Scenes/NPC/Enemy/Enemy.gd

```diff
@@ -14,12 +14,15 @@ func _physics_process(_delta):
 	velocity = direction * movementSpeed
 	move_and_slide()
 
-func DamageEnemy(damage):
+func DamageEnemy(damage, incoming_dmg_pos):
```

Comment:
I think we can add the knockback value here as an agrument and then have all the sub functions call this one.